frontend-react/src/components/MessageTracker/MessageDetails.tsx

```diff
@@ -15,57 +18,154 @@ type MessageDetailsProps = {
     id: string | undefined;
 };
 
+const parseFileLocation = (
+    fileLocation: string
+): {
+    fileLocation: string;
+    sendingOrg: string;
+    fileName: string;
+} => {
+    let fileLocationFromStr = "";
+    let sendingOrgFromStr = "";
+    let fileNameFromStr = "";
+
+    let fileReportsLocation = fileLocation.split("reports/").pop();
+    if (fileLocation.includes("reports/") && fileReportsLocation) {
```

Comment:
Do we want to display the filename even if "reports/" is not included in the fileLocation?

Reply:
I'd love to learn more about this function and it's intended design. It seems to rely on the formatting of the url with `%2F` to indicate: location, sending organization and file name in that order. Is there a way we can restructure the backend response to form this into URL parameters? Then we could at least use `new URLSearchParams` to parse out the data much more easily / type it.

In regards specifically to `display the filename even if "reports/" is not included` I would think so, but what does that URL look like? Just the sending org + the filename? Is the data ALWAYS structured that 3 part URL === fileLocation, sendingOrg and fileName and 2 part URL === fileLocation and sendingOrg?

Reply:
@stephenkao thoughts on this?

Reply:
Good point!  There's definitely some cleanup we can do with this, but I think we can assume this format will be okay for the time being -- as far as I know, the structure of this URL is pretty set (at least with the COVID pipeline) since that signifies the step in the pipeline.  We could also honestly probably just grab whatever's after the last unencoded slash instead of splitting on `reports` 🤷

Reply:
@stephenkao  I'm splitting on `reports/ `since I need to grab the folder location, sending org, and filename which all come after `reports/ `. 
If I grab whatever's after the last unencoded slash, I will only get the filename.

Reply:
Hmmmm isn't the URL structure like this?

```
https://azurite:10000/devstoreaccount1/reports/receive%2Fsimple_report.csvuploader%2Fupload-covid-19-c33f9d36-9e5b-44eb-9368-218d88f3a7d1-20230131190253.csv
```

If we grab the last fragment when splitting on `'/'`, we should get `receive%2Fsimple_report.csvuploader%2Fupload-covid-19-c33f9d36-9e5b-44eb-9368-218d88f3a7d1-20230131190253.csv` -- then we can just split on the encoded `'%2F'` to get the folder location, Sender, and filename, right?

Reply:
Yup, sorry I misread what you meant previously. Will update.